x/group/server/msg_server.go

```diff
@@ -106,17 +114,14 @@ func (s serverImpl) UpdateGroupMembers(ctx types.Context, req *group.MsgUpdateGr
 				return sdkerrors.Wrap(err, "get group member")
 			}
 
-			totalWeight, err := math.ParseNonNegativeDecimal(g.TotalWeight)
-			if err != nil {
-				return err
-			}
 			newMemberWeight, err := math.ParseNonNegativeDecimal(groupMember.Member.Weight)
 			if err != nil {
 				return err
 			}
 
-			// handle delete
-			if newMemberWeight.Cmp(apd.New(0, 0)) == 0 {
+			// Handle delete for members with zero weight.
+			if newMemberWeight.IsZero() {
+				// We can't delete a group member that doesn't already exist.
 				if !found {
 					return sdkerrors.Wrap(orm.ErrNotFound, "unknown member")
```

Comment:
There's no harm in ignoring this case right?

Reply:
yeah I guess `groupMemberTable.Delete` below would just fail in this case, but that would be an unnecessary access to the store given that we already know that it should fail, wdyt?

Reply:
We just wouldn't even execute this block... but maybe let's keep it as is because we would need to adjust tests and I don't want to further churn on this PR.